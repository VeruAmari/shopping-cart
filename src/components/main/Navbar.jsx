import styled from 'styled-components';
import PropTypes from 'prop-types';
// import { Link } from 'react-router-dom';
import { NavElement, ExNavElement } from '../sub/Containers';
import Logo from '../sub/Logo';
import cart from './cart.svg';

const Navbar = ({ cartProducts }) => {
  let amountOfProducts = 0;
  if (cartProducts) {
    for (let product in cartProducts) {
      amountOfProducts += cartProducts[product];
    }
  }

  return (
    <Wrapper as="nav">
      <NavElement to="/">
        <Logo />
      </NavElement>
      <ExNavElement to="shopping/shop">To Shop</ExNavElement>

      <NavElement to="shopping/checkout">
        <CartWrapper>
          Cart
          <StyledCart />
          {cartProducts && <AmountCircle> {amountOfProducts} </AmountCircle>}
        </CartWrapper>
      </NavElement>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  position: sticky;
  top: 0;
  align-self: start;
  grid-template-columns: auto auto auto;
  display: grid;
  justify-content: space-around;
  background-color: var(--color-3);
`;

const CartWrapper = styled.div`
  display: grid;
  grid-template-columns: auto auto auto;
  position: relative;
`;
const AmountCircle = styled.div`
  position: absolute;
  top: -30%;
  right: -30%;
  background-color: var(--color-2);
  color: var(--color-5);
  display: grid;
  justify-items: center;
  justify-content: center;
  padding: 0;
  margin: 0;
  text-align: center;
  align-content: center;
  align-items: center;
  width: 1.2rem;
  height: 1.2rem;
  border-radius: 99%;
`;

const StyledCart = styled.div`
  background-image: url(${cart});
  width: 20px;
  height: 20px;
`;

Navbar.propTypes = {
  cartProducts: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
};

export default Navbar;
