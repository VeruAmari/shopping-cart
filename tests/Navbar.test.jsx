import { render, screen } from '@testing-library/react';

import Navbar from '../src/components/main/Navbar';

describe('Navbar', () => {
  it('renders', () => {
    render(<Navbar>Header</Navbar>);

    expect(screen.getByText('Header').textContent).toMatch(/header/i);
  });
});
