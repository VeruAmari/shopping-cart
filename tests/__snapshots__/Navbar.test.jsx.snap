// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > matches snapshot 1`] = `
<nav
  class="Navbar__Wrapper-sc-uyr9r3-0 hzwQAV"
>
  <a
    class="Containers__NavElement-sc-1lz7g2b-1 Containers__ExNavElement-sc-1lz7g2b-2 fyMQmt khIBjc"
    href="/home"
  >
    <span
      class="Logo__LogoWrapper-sc-1s45uey-1 jSojiy"
      size="1"
    >
      Scui
    </span>
  </a>
  <a
    class="Containers__NavElement-sc-1lz7g2b-1 Containers__ExNavElement-sc-1lz7g2b-2 fyMQmt khIBjc"
    href="/shopping/shop"
  >
    To Shop
  </a>
  <a
    class="Containers__NavElement-sc-1lz7g2b-1 Containers__ExNavElement-sc-1lz7g2b-2 fyMQmt khIBjc"
    href="/shopping/checkout"
  >
    <div
      class="Navbar__CartWrapper-sc-uyr9r3-1 blSLUB"
    >
      Cart
      <img
        class="Navbar__StyledCart-sc-uyr9r3-3 jGcCaT"
        src="/src/assets/cart.svg"
      />
    </div>
  </a>
</nav>
`;
